cmake/linker_script/arm/linker.cmake

```diff
@@ -96,7 +96,7 @@ if(CONFIG_CPP)
   zephyr_linker_section_configure(SECTION .ARM.extab INPUT ".gnu.linkonce.armextab.*")
 endif()
 
-zephyr_linker_section(NAME .ARM.exidx GROUP ROM_REGION)
+zephyr_linker_section(NAME .ARM.exidx GROUP TEXT_REGION)
```

Comment:
In my application, I need to move data to an external memory-mapped flash because the internal one is full. I can't move `.text` because breakpoints only work in internal flash (FPB revision 1), but that also means I can't move `.ARM.exidx` because it has `R_ARM_PREL31` relocations into `.text`, which would be out of range in the external flash. With all the patches here, the solution is pretty straightforward, I just add the following to my `CMakeLists.txt`:

```cmake
zephyr_linker_group(NAME ROM_REGION LMA QSPI)
zephyr_linker_group(NAME TEXT_REGION LMA FLASH SYMBOL SECTION)
zephyr_linker_section(NAME .rom_start ADDRESS ${FLASH_ADDR} GROUP TEXT_REGION NOINPUT)
```

Reply:
But doesn't the commit: 372ac65365a985edb8edd94e722103d6be66823a allows you to redefine `.ARM.exidx` externally, and thus inside your app's CMakeLists.txt do:
```
zephyr_linker_group(NAME ROM_REGION LMA QSPI)
zephyr_linker_group(NAME TEXT_REGION LMA FLASH SYMBOL SECTION)
zephyr_linker_section(NAME .rom_start ADDRESS ${FLASH_ADDR} GROUP TEXT_REGION NOINPUT)
zephyr_linker_section(NAME .ARM.exidx GROUP TEXT_REGION)
```
or (to avoid messing with the text section) 
```
zephyr_linker_section(NAME .ARM.exidx  LMA FLASH)
```

?

Reply:
Sure, I can do that and remove that commit from this PR.